rusoto/services/sts/src/custom/credential.rs

```diff
@@ -195,6 +195,12 @@ impl StsSessionCredentialsProvider {
             inner: self.sts_client.get_session_token(request),
         }
     }
+
+    /// Convert the credential provider to one that caches the result instead of
+    /// making a call every time credentials are needed.
+    pub fn to_auto_refresh_credentials(self) -> Result<AutoRefreshingProvider<StsSessionCredentialsProvider>, CredentialsError> {
```

Comment:
The [API guidelines](https://rust-lang-nursery.github.io/api-guidelines/naming.html) suggest that this method should be called `into_*` to indicate the fact that it is taking ownership of `self`.

While I'm nitpicking, you can use `Self` here instead of `StsSessionCredentialsProvider` to add a little more brevity to the signature.